mobile/src/main/java/org/openhab/habdroid/background/BackgroundUtils.java

```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * Makes notification for background tasks. Sets notification channel, importance and more
+     * depending on the parameters.
+     *
+     * @param context
+     * @param msg Message to show.
+     * @param iconId Icon to show.
+     * @param isError true if it's an error.
+     * @param action Action to show as a button, e.g. "Retry"
+     * @return
+     */
+    public static Notification makeBackgroundNotification(Context context, String msg,
+            @DrawableRes int iconId, boolean isError, NotificationCompat.Action action) {
+        Intent notificationIntent = new Intent(context, MainActivity.class);
+        notificationIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP
+                | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        PendingIntent intent = PendingIntent.getActivity(context, 0,
+                notificationIntent, 0);
+
+        NotificationCompat.Builder nb = new NotificationCompat.Builder(context,
+                isError ? CHANNEL_ID_BACKGROUND_ERROR : CHANNEL_ID_BACKGROUND)
+                .setLargeIcon(drawableToBitmap(context.getResources().getDrawable(iconId)))
+                .setSmallIcon(R.drawable.ic_openhab_appicon_white_24dp)
+                .setContentTitle(context.getString(R.string.app_name))
+                .setWhen(System.currentTimeMillis())
+                .setStyle(new NotificationCompat.BigTextStyle().bigText(msg))
+                .setContentText(msg)
+                .setAutoCancel(true)
+                .setContentIntent(intent)
+                .setColor(ContextCompat.getColor(context, R.color.openhab_orange))
+                .setCategory(isError ? NotificationCompat.CATEGORY_ERROR
+                        : NotificationCompat.CATEGORY_PROGRESS)
+                .setOngoing(!isError)
+                .setGroup(isError ? CHANNEL_ID_BACKGROUND_ERROR : CHANNEL_ID_BACKGROUND)
+                .setPriority(isError ? NotificationCompat.PRIORITY_DEFAULT
+                        : NotificationCompat.PRIORITY_LOW);
+
+        if (isError) {
+            nb.setLights(ContextCompat.getColor(context, R.color.openhab_orange),
+                    3000, 3000);
+        }
+
+        if (action != null) {
+            nb.addAction(action);
+        }
+
+        return nb.build();
+    }
+
+    /**
+     * Makes a "Retry" action.
+     * @param context
+     * @param notificationId Id of the current notification
+     * @return
+     */
+    public static NotificationCompat.Action makeRetryAction(Context context, int notificationId) {
+        Intent retryIntent = new Intent(context, BackgroundTaskRetryBroadcastReceiver.class);
+        retryIntent.putExtra(OH_EXTRA_NOTIFICATION_ID, notificationId);
+        PendingIntent retryPendingIntent =
+                PendingIntent.getBroadcast(context, 0, retryIntent, 0);
+        return new NotificationCompat.Action(
+                R.drawable.ic_refresh_grey_24dp,
+                context.getString(R.string.error_sending_alarm_clock_action_retry),
```

Comment:
The method is generic, but the string resource name isn't ... should probably adapt the latter.

Reply:
Done